src/panels/config/js/trigger/device.tsx

```diff
@@ -41,9 +68,68 @@ export default class DeviceTrigger extends Component<any, any> {
           hass={hass}
           label="Trigger"
         />
+        {showFor && (
+          <paper-time-input
+            label={hass.localize(
+              "ui.panel.config.automation.editor.triggers.type.state.for"
+            )}
+            hour={hours}
+            min={minutes}
+            sec={seconds}
+            enable-second
+            format={24}
+            onhour-changed={this.hourChanged}
```

Comment:
Can we make `paper-input-time` fire a `change` with the new value if any of the inputs are changed, that prevents us from having to do formatting and listening twice.

Reply:
Remove all the listeners for hour, minute and seconds and instead listen for value:
````
    onvalue-changed={this.timeChanged}
```
We then get the formatted time string in the correct format, so need to have `_formatTime` and pad in `device.tsx`.

```
  public timeChanged(ev) {
    if (!this.props.trigger.hasOwnProperty("for")) {
      return;
    }
    this.props.onChange(this.props.index, {
      ...this.props.trigger,
      for: ev.detail.value,
    });
  }
```

Reply:
I started like that, but it became annoying because the `paper-input-time`'s value is `undefined` if one of hour, minute or second is empty. Will give it another go though.

Reply:
Oh... we might want to change that in `paper-input-time`? empty field = 0?

Reply:
OK, I gave it a try.